install/index.php

```diff
@@ -1,276 +1,419 @@
-        $db_port = ($_POST['db_type'] == 'mysql' && $_POST['db_port'] != 3306)
-            || ($_POST['db_type'] == 'pgsql' && $_POST['db_port'] != 5432) ? $_POST['db_port'] : '';
-
-        // For SQLite we should provide path to db file
-        $db_file = $_POST['db_type'] =='sqlite' ? $db_file = $_POST['db_name'] : '';
-
-        echo $twig->render('step6.html', array(
-            'next_step' => ++$current_step,
-            'language' => $language,
-            'config_file_created' => $config_file_created,
-            'local_config_file' => LOCAL_CONFIG_FILE,
-            'session_key' => Poweradmin\Password::salt(SESSION_KEY_LENGTH),
-            'iface_lang' => $language,
-            'dns_hostmaster' => $_POST['dns_hostmaster'],
-            'dns_ns1' => $_POST['dns_ns1'],
-            'dns_ns2' => $_POST['dns_ns2'],
-            'db_host' => $_POST['db_host'],
-            'db_user' => $_POST['pa_db_user'],
-            'db_pass' => $_POST['pa_db_pass'],
-            'db_name' => $_POST['db_name'],
-            'db_type' => $_POST['db_type'],
-            'db_port' => $db_port,
-            'db_charset' => $_POST['db_charset'],
-            'pa_pass' => $_POST['pa_pass']
-        ));
-        break;
+        // Create default permission-template
+        $permissionTemplateResult = $dbConnection->executeQuery(
+            $queryBuilder->insert('perm_templ')->values(['name' => '?', 'descr' => '?']),
+            ['Administrator', _('Administrator template with full rights.')]
+        );
 
-    case 7:
-        echo $twig->render('step7.html');
-        break;
+        $permissionTemplateItemResult = $dbConnection->executeQuery(
+            $queryBuilder->insert('perm_templ_items')->values(['templ_id' => '?', 'perm_id' => '?']),
+            [1, 53]
+        );
+
+
+        // Get user-data
+        $user = $_SESSION['userData'];
 
-    default:
+        // Replace password with hashed password
+        $user['password'] = Poweradmin\Password::hash($user['password']);
+
+        // Create user-query
+        $userQuery = $queryBuilder->insert('users')->values([
+            'username' => '?', 'password' => '?', 'fullname' => '?', 'email' => '?', 'description' => '?',
+            'perm_templ' => '?', 'use_ldap' => '?', 'active' => '?'
+        ]);
+
+        // Insert user
+        $userResult = $dbConnection->executeQuery(
+            $userQuery,
+            [
+                $user['username'], $user['password'], $user['fullname'], $user['email'],
+                _('Administrator with full rights.'), 1, false, true
+            ]
+        );
+
+
+        // Get template of config-file
+        $configFileContent = file_get_contents(INSTALLER_DIRECTORY . 'includes/config_template.php');
+
+        // Get db-parameters & general-data from session
+        $dbParameters = $_SESSION['dbParameters'];
+        $generalData = $_SESSION['generalData'];
+
+        // Replace placeholder
+        $configFileContent = str_replace([
```

Comment:
maybe some parts of code can be extracted to includes/functions.php ? For example building of configuration file or creating db tables, it would be good to have less code in installer steps